### PR TITLE
Initial mui dropdown textarea added

### PR DESCRIPTION
**Pull request by aritraibm**

### CODE REVIEW

File: src/Constants/application.type.ts [hunk 1]
```diff
@@ -2,3 +2,46 @@ export interface SinleMenuListHOC {
...
+    projectBandwidthAvailble: string;
+    leftMenuHome: string;
+    leftMenuILC: string;
+    leftMenuForum: string;
+    leftMenuProjectDetails: string;
+    leftMenuAdmin: string;
+    leftMenuSettings: string;
+    isRequired: string;
+    maxMessage: string;
+}
```
Comment (PramodaM4444) [inline src/Constants/application.type.ts]: Is this required to specify the types for the UI Constants?

File: src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const selectDefaultProps = {
+    variant: "filled",
+    fullWidth: true,
+    textcolor: "black",
+    label: "inputText",
+};
+
+export const Dropdown: React.FC<any> = React.forwardRef(
+    (props: DropdownType, ref) => (
+        <FormControl
```
Comment (PramodaM4444) [inline src/components/Dropdown/Dropdown.tsx]: Can we destructure the props?

Ex :  const { helperText, options, label } = props;

File: src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            variant="filled"
+            {...formControlDefaultProps}
+            error={props.helperText}
+        >
+            <InputLabel id={props.name}>{props.label}</InputLabel>
+            <SelectInput {...selectDefaultProps} {...props} ref={ref}>
+                <MenuItem value="">
+                    <em>Select an option</em>
```
Comment (PramodaM4444) [inline src/components/Dropdown/Dropdown.tsx]: Pls move this to Constants

File: src/containers/AddProjectDetails/AddProjectDetailsForm.tsx [hunk 4]
```diff
@@ -29,13 +40,113 @@ export default function AddProjectDetailsForm() {
...
-                    helperText={errors.email ? errors?.email.message : null}
+                    label={UIConstants.projectCode}
+                    {...register("projectCode")}
+                    error={!!errors?.projectCode}
+                    helperText={
+                        errors.projectCode ? errors?.projectCode.message : null
+                    }
+                />
+            </styledComp.flexRow>
+            <styledComp.flexRow>
```
Comment (PramodaM4444) [inline src/containers/AddProjectDetails/AddProjectDetailsForm.tsx]: Can we import the Styled components similar to 
import { **FlexRow**, **FlexAttribute** } from "./AddProjectDetailsForm.style";

Note: Components name should start with the Capital letter.

File: src/containers/AddProjectDetails/AddProjectDetailsForm.tsx [hunk 5]
```diff
@@ -29,13 +40,113 @@ export default function AddProjectDetailsForm() {
...
+                        error={!!errors?.projectCompletionDate}
+                        onChange={handleChange}
+                        options={channelOwnerLabelValues}
+                        helperText={
+                            errors.projectCompletionDate
+                                ? errors?.projectCompletionDate.message
+                                : null
+                        }
+                    /> 
+                </Grid> */}
```
Comment (PramodaM4444) [inline src/containers/AddProjectDetails/AddProjectDetailsForm.tsx]: Pls remove the commented out code

File: src/Constants/application.type.ts (hunk 1 above)
Comment (aritraibm) [reply src/Constants/application.type.ts]: It's always better to have strong type safety, pls refer **src/Constants/UIConstants.tsx** or **src/utils/validation-helper.ts**, however, if you recommend, we can remove it. Pls suggest.

File: src/components/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (aritraibm) [reply src/components/Dropdown/Dropdown.tsx]: Agreed! will change it.

File: src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (aritraibm) [reply src/components/Dropdown/Dropdown.tsx]: Hmm... my bad! :( will change it

File: src/containers/AddProjectDetails/AddProjectDetailsForm.tsx (hunk 5 above)
Comment (aritraibm) [reply src/containers/AddProjectDetails/AddProjectDetailsForm.tsx]: This structure is partially required, will replace this with the date picker attribute next to the next PR

File: src/containers/AddProjectDetails/AddProjectDetailsForm.tsx (hunk 4 above)
Comment (aritraibm) [reply src/containers/AddProjectDetails/AddProjectDetailsForm.tsx]: All right, but I have a doubt. will not be something like we'll be importing more when the number of CSS will be higher? i.e.:

import { CSS1, CSS2, CSS3, CSS4, CSS5, CSS6, CSS7, CSS8, CSS9, CSS10} from "./Something.style";

Pls suggest.

File: src/containers/AddProjectDetails/AddProjectDetailsForm.tsx (hunk 5 above)
Comment (aritraibm) [reply src/containers/AddProjectDetails/AddProjectDetailsForm.tsx]: Okay will remove it :)

File: src/Constants/application.type.ts (hunk 1 above)
Comment (PramodaM4444) [reply src/Constants/application.type.ts]: Strong type safety is required. However, in general, we are storing strings in the Constants file and also we are not trying to alter the Constants value after importing it into other files. Hence it's not absolutely required to specify types for Constants.

File: src/containers/AddProjectDetails/AddProjectDetailsForm.tsx (hunk 4 above)
Comment (PramodaM4444) [reply src/containers/AddProjectDetails/AddProjectDetailsForm.tsx]: Yes, we will have more named imports when the styles increases. However, we are also splitting the larger components into smaller pieces by extracting them into a separate component file. Also, we are trying to maintain consistency.

Comment (aritraibm) [reply src/containers/AddProjectDetails/AddProjectDetailsForm.tsx]: okay

File: src/Constants/application.type.ts (hunk 1 above)
Comment (aritraibm) [reply src/Constants/application.type.ts]: okay

File: src/containers/AddProjectDetails/AddProjectDetailsForm.tsx (hunk 5 above)
Comment (aritraibm) [reply src/containers/AddProjectDetails/AddProjectDetailsForm.tsx]: done

File: src/containers/AddProjectDetails/AddProjectDetailsForm.tsx (hunk 4 above)
Comment (aritraibm) [reply src/containers/AddProjectDetails/AddProjectDetailsForm.tsx]: done

File: src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (aritraibm) [reply src/components/Dropdown/Dropdown.tsx]: done

File: src/components/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (aritraibm) [reply src/components/Dropdown/Dropdown.tsx]: done

File: src/Constants/application.type.ts (hunk 1 above)
Comment (aritraibm) [reply src/Constants/application.type.ts]: done

**Review by PramodaM4444 — APPROVED** [review]

**Pull request merged**